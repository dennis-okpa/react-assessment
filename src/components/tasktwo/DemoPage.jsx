import React, { useReducer } from 'react';
import {
  Content,
  Heading,
  Task,
  Modal,
  Overlay,
  DemoButton,
} from '@components/styled-components/Task';
import { modalSelection, initialState } from '../../utils/redux/reducer';
import { ACTION } from '../../assets/constants';

const DemoPage = () => {
  const [state, dispatch] = useReducer(modalSelection, initialState);

  return (
    <Task>
      <Heading>Demo Page</Heading>
      <Content>
        <DemoButton type="button" onClick={() => dispatch({ type: ACTION.DATA_LOADER })}>Data loading modal</DemoButton>
        <DemoButton type="button" onClick={() => dispatch({ type: ACTION.DELETE_FILE })}>Delete file modal</DemoButton>
        <DemoButton type="button" onClick={() => dispatch({ type: ACTION.DELETE_REPORT })}>Delete report modal</DemoButton>
      </Content>
      {state.modal && (
        <>
          <Modal>{state.modal}</Modal>
          <Overlay onClick={() => dispatch({ type: 'reset' })} />
        </>
      )}
    </Task>
  );
};

export default DemoPage;
