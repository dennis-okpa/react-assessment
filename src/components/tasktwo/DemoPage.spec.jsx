import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import DemoPage from './DemoPage';

jest.mock('@components/ui-components/DeleteFileModal', () => ({
  __esModule: true,
  // eslint-disable-next-line react/display-name
  default: () => <div data-testid="delete_file_modal" />
}));

jest.mock('@components/ui-components/DeleteReportModal', () => ({
  __esModule: true,
  // eslint-disable-next-line react/display-name
  default: () => <div data-testid="delete_report_modal" />
}));

afterEach(() => {
  jest.clearAllMocks();
});

describe('The DemoPage component', () => {
  describe('when renders', () => {
    let container;
    let getByText;
    let getByTestId;
    let getByAltText;

    beforeEach(() => {
      ({
        container, getByText, getByTestId, getByAltText
      } = render(<DemoPage />));
    });

    it('should match snapshot', () => {
      expect(container).toMatchSnapshot();
    });

    describe('when the show data loader modal button is clicked', () => {
      it('should dispatch the data loader modal', () => {
        fireEvent.click(getByText('Data loading modal'));

        expect(getByAltText('data loader')).toBeInTheDocument();
      });
    });

    describe('when the show delete file modal button is clicked', () => {
      it('should dispatch the data loader modal', () => {
        fireEvent.click(getByText('Delete file modal'));

        expect(getByTestId('delete_file_modal')).toBeInTheDocument();
      });
    });

    describe('when the show delete report modal button is clicked', () => {
      it('should dispatch the data loader modal', () => {
        fireEvent.click(getByText('Delete report modal'));

        expect(getByTestId('delete_report_modal')).toBeInTheDocument();
      });
    });
  });
});
