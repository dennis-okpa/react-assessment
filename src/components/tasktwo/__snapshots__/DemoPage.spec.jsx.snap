// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The DemoPage component when renders should match snapshot 1`] = `
<div>
  <div
    class="Task-sc-7z955n-0 lgLqQs"
  >
    <h1
      class="Task__Heading-sc-7z955n-3 dFWZeR"
    >
      Demo Page
    </h1>
    <div
      class="Task__Content-sc-7z955n-1 jkQdat"
    >
      <button
        class="Task__DemoButton-sc-7z955n-8 gFvSgS"
        type="button"
      >
        Data loading modal
      </button>
      <button
        class="Task__DemoButton-sc-7z955n-8 gFvSgS"
        type="button"
      >
        Delete file modal
      </button>
      <button
        class="Task__DemoButton-sc-7z955n-8 gFvSgS"
        type="button"
      >
        Delete report modal
      </button>
    </div>
  </div>
</div>
`;
